### ISSUE TITLE
CAN AutoRetransmission setting logic is inverted

### ISSUE THREAD

**Issue by julian-poidevin** [foxBMS/foxbms#11]
In the following lines, documentation and setting are incorrect :
https://github.com/foxBMS/foxbms/blob/ca7e8d58e8bf26559b63c53bc61c5b18f48a0347/embedded-software/mcu-primary/src/driver/config/can_cfg.c#L179-L181

https://github.com/foxBMS/foxbms/blob/ca7e8d58e8bf26559b63c53bc61c5b18f48a0347/embedded-software/mcu-primary/src/driver/config/can_cfg.c#L296-L298

Indeed, `AutoRetransmission` field should be set to :
- `ENABLE` : if we want to retransmit the message until it has been successfully transmitted
- `DISABLE` : to transmit only once, independently of transmission result

This error may has been caused by the CAN HAL driver update which changes the logic of this field (previously `NART`). See below this message on ST Forum : https://community.st.com/s/question/0D50X00009Xkg2MSAR/stm32f4xx-rework-of-hal-can-driver-examples

![image](https://user-images.githubusercontent.com/8756421/104735887-74903d00-5742-11eb-8510-b91be65ce0ae.png)

**Comment by foxBMS** [comment]
Dear @julian-poidevin,

thanks for sharing, we will investigate this closely.
We come back as soon we have new information.

Best regards,
The foxBMS Team

**Comment by julian-poidevin** [comment]
@foxBMS I submitted a Pull Request to fix the issue : https://github.com/foxBMS/foxbms/pull/12

**Comment by julian-poidevin** [comment]
Hi @foxBMS did you investigate the issue ?

**Comment by julian-poidevin** [comment]
Hi again @foxBMS, did you look into the issue ?

**Comment by foxBMS** [comment]
Dear @julian-poidevin,

you are perfectly right, we could reproduce erroneous behaviour on the CAN bus based on this setting, therefore this is clearly a bug. We will make a bug fix release tomorrow (v1.6.6), that also includes a fix for this.

Many thanks for pointing us at this. However we are currently not able to accept Pull Requests on GitHub due to a missing Contributors License Agreement (CLA) system. I will add a link (to this issue) that shows the new Pull Request template (also released tomorrow) that explains this further.

We would still be very pleased if you intend to contribute to this project.
Please contact us first at info@foxbms.org in order to discuss your contribution and how we can apply your contributions to the project for future inputs.

Thanks again, for pointing us at this bug.

Best regards,
The foxBMS Team

**Comment by foxBMS** [comment]
Dear @julian-poidevin,

we have released v1.6.6 that also includes this fix (see [v1.6.6](https://github.com/foxBMS/foxbms/releases/tag/v1.6.6))

Please see the new Pull requests template ([.github/PULL_REQUEST_TEMPLATE.md](https://github.com/foxBMS/foxbms/blob/c5fb868b59468e59d5b47f56e72f46bb276e2a91/.github/PULL_REQUEST_TEMPLATE.md)) for more details why we currently can not accept Pull requests.

Best regards,
The foxBMS Team

**Comment by julian-poidevin** [comment]
GReat

> Dear @julian-poidevin,
> 
> we have released v1.6.6 that also includes this fix (see [v1.6.6](https://github.com/foxBMS/foxbms/releases/tag/v1.6.6))
> 
> Please see the new Pull requests template ([.github/PULL_REQUEST_TEMPLATE.md](https://github.com/foxBMS/foxbms/blob/c5fb868b59468e59d5b47f56e72f46bb276e2a91/.github/PULL_REQUEST_TEMPLATE.md)) for more details why we currently can not accept Pull requests.
> 
> Best regards,
> The foxBMS Team

Great, I've seen other bugs, I'll submit them directly to the email address you provided !
